src/components/DatePicker/DatePicker.js

```diff
@@ -97,10 +97,24 @@ export default class DatePicker extends Component {
         cal.calendarContainer.focus();
       }
     });
+    input.addEventListener('change', () => {
```

Comment:
Given this line deals with DOM directly, the event listeners need to be be removed when the React component is unmounted. [`carbon-components/es/globals/js/misc/on`](https://github.com/carbon-design-system/carbon-components/blob/a4b23e4/src/globals/js/misc/on.js) may help - e.g. `this.hChange = on(input, 'change', () => { … })` when mounting and `if (this.hChange) { this.hChange = this.hChange.release(); }` when unmounting.

Reply:
Ahh right, I'll make the change, thanks!